plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ApplyAclRuleVspCommand.java

```diff
@@ -79,4 +191,47 @@ public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ApplyAclRuleVspCommand that = (ApplyAclRuleVspCommand) o;
+
+        if (_egressDefaultPolicy != that._egressDefaultPolicy) return false;
+        if (_isL2Network != that._isL2Network) return false;
+        if (_networkAcl != that._networkAcl) return false;
+        if (_networkId != that._networkId) return false;
+        if (_networkReset != that._networkReset) return false;
+        if (_aclRules != null ? !_aclRules.equals(that._aclRules) : that._aclRules != null) return false;
+        if (_acsIngressAcl != null ? !_acsIngressAcl.equals(that._acsIngressAcl) : that._acsIngressAcl != null)
+            return false;
+        if (_domainTemplateName != null ? !_domainTemplateName.equals(that._domainTemplateName) : that._domainTemplateName != null)
+            return false;
+        if (_networkDomainUuid != null ? !_networkDomainUuid.equals(that._networkDomainUuid) : that._networkDomainUuid != null)
+            return false;
+        if (_networkName != null ? !_networkName.equals(that._networkName) : that._networkName != null) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_vpcOrSubnetUuid != null ? !_vpcOrSubnetUuid.equals(that._vpcOrSubnetUuid) : that._vpcOrSubnetUuid != null)
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This ``equals`` implementation should include attributes from the ``Command`` super class.